text/0002-problems-with-xul.md

```diff
@@ -0,0 +1,69 @@
+# XUL problems
+
+XUL is often mentioned as a source of problems for developers. This document
+aims to list the different kinds of problems that exist.
+
+## XUL ownership
+
+XUL is largely unowned. While there are developers who know and understand the
+XUL code they are few and mostly dedicated to other projects at present.
+
+## XUL bugs
+
+XUL contains bugs. This isn’t due to anything particularly special about XUL but
+when we encounter bugs we frequently choose to not fix them, instead trying to
+find workarounds in the UI instead. See for example
+https://bugzilla.mozilla.org/show_bug.cgi?id=354527.
+
+## No new features
+
+There is no new feature implementation going on in XUL compared to HTML which
+receives constant improvements to meet the demands of the web.
+
+## Performance issues
+
+It is thought that XUL’s performance is worse than that of HTML where we have spent
+large amounts of time optimising for the web but little amount of time optimising
+for XUL.
```

Comment:
There is a counter argument to this which we should perhaps acknowledge; XUL layout is designed for user interfaces where elements do not generally flow like text does. HTML flexbox (a natural replacement for XUL layout) _is_ designed for document layout, and therefore the algorithms required to implement HTML flexbox are considerably more compute intensive than the XUL equivalent, which can lead to cases where XUL is inherently faster even without optimization.

Also a counter-counter  argument - It is possible that grid layout is a better alternative than flexbox, with better performance characteristics. I'm not sure of the detail.

Either way I don't think this nullifies the performance point.